cluster/addons/addon-manager/kube-addons-main.sh

```diff
@@ -35,18 +35,13 @@ fi
 log INFO "== Kubernetes addon manager started at $(date -Is) with ADDON_CHECK_INTERVAL_SEC=${ADDON_CHECK_INTERVAL_SEC} =="
 
 # Wait for the default service account to be created in the kube-system namespace.
-token_found=""
-while [ -z "${token_found}" ]; do
+# shellcheck disable=SC2086
+# Disabling because "${KUBECTL_OPTS}" needs to allow for expansion here
+while [[ $(${KUBECTL} ${KUBECTL_OPTS} get --namespace="${SYSTEM_NAMESPACE}" serviceaccount default -o jsonpath='{.metadata.name}') != "default" ]]; do
```

Comment:
this looks weird... we don't need the jsonpath bit, we just need the lookup to not 404

Reply:
the output for 404 is "Error from server (NotFound): serviceaccounts "defaulta" not found
".  the other way is do substring matching for `NotFound`. using jsonpath might be a stronger assertion.

Reply:
the exit code is non-zero, you can loop on that

Reply:
something like

```shell
while ! ${KUBECTL} ${KUBECTL_OPTS} get --namespace="${SYSTEM_NAMESPACE}" serviceaccount default2; do
  echo "== Error getting default service account, retry in 0.5 second =="
  sleep 0.5
done
```

Reply:
forgot the exit code. done.